src/types/UserData.ts

```diff
@@ -0,0 +1,5 @@
+import { DialogData } from "./DialogData";
+
+export type UserData = {
+  [id: string]: DialogData;
```

Comment:
Will this make it easier for understand the intent of `id`?

```suggestion
  [dialogId: string]: DialogData;
```